service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderControllerMidEventTest.java

```diff
@@ -598,4 +603,37 @@ private DocumentReference expectedDraftDocument() {
             return childElement;
         }
     }
+
+    @Nested
+    class PopulateEPOTypeAndAddressMidEvent {
+
+        private final String callbackType = "populate-epo-parameters";
+
+        @Test
+        void shouldPrePopulateEpoOrderFieldsIfPresentInCaseData() {
+            AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(
+                buildCaseData(REMOVE_TO_ACCOMMODATION), callbackType);
+
+            CaseData caseData = extractCaseData(callbackResponse);
+
+            assertThat(caseData.getOrders().getEpoType().equals(REMOVE_TO_ACCOMMODATION));
+        }
+
+        @Test
+        void shouldPrePopulateAddressFieldsIfPresentInCaseData() {
+            AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(
+                buildCaseData(PREVENT_REMOVAL), callbackType);
+
+            CaseData caseData = extractCaseData(callbackResponse);
+
+            assertThat(caseData.getOrders().getEpoType().equals(PREVENT_REMOVAL));
+        }
+
+        private CaseData buildCaseData(EPOType epoType) {
```

Comment:
maybe we need another test to assert that excluded is prepopulated also

Reply:
sorted